### PR TITLE
Added create objects, update objects, home and map code structure

### PR DESCRIPTION
**Pull request by JohanDjarvKarltorp**
- closes issues: #48 #49 (#52?)
- map.html js files are rewritten with classes instead to simplify the code for easier reading
- Added comments to all functions in js files
- Added newObject.html and newObj.js
- Added updateObject.html and updateObj.js
- Added home.html and home.js
- Updated some styling to most HTML files
- Added navbar to most HTML pages
- On map.html, products are loaded from database and data is loaded and saved from/to backend
- image files were renamed to match Modell name (in future images should be loaded from base64 instead)
- some file structure were added
- removed exampleSaveData.json 
- removed jsonSave.json

### CODE REVIEW

File: js/map/popup.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	    Personer per hushåll: <input type="text" name="per" value="${nop}"><br>
+	    Vatten per person/dygn: <input type="text" name="cons" value="${flow}"><br>
+	    <input type="button" value="Ändra">
+	    </form>
+	    </div>`;
+    },
+
+    pipe: (dimension, tilt) => {
+        return `<b>Rör</b><br>
+<label>Inner Dimension</label>
```
Comment (eskoglund) [inline js/map/popup.js]: Ändra till "Innediameter"

File: js/map/loadLeafletMaterialMap.js [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+*/
+/**
+ * load - Load objects(markers, polylines, polygons) to the map using json
+ * data
+ *
+ * @returns {void}
+ **/
+let load = () => {
+    const savedData = jsonData;
+    //const jsonLoad = JSON.parse(jsonData)
```
Comment (eskoglund) [inline js/map/loadLeafletMaterialMap.js]: Ta bort ifall den inte används

File: calculations.html [hunk 3]
```diff
@@ -8,6 +8,12 @@
 		<link rel="stylesheet" href="css/calculations.css">
 	</head>
 	<body>
+		<div class="topnav">
+			<a href="home.html">Home</a>
```
Comment (eskoglund) [inline calculations.html]: Ändra till "Hem" i texten eftersom vi kör allt annat på svenska

File: home.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+		<title>Home</title>
+		<link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+		<link href="https://fonts.googleapis.com/css?family=Fira+Sans:300,400" rel="stylesheet">
+		<link rel="stylesheet" href="css/home.css">
+		<script src="js/configVariables.js" charset="utf-8"></script>
+		<script src="js/projects/home.js"></script>
+	</head>
+	<body>
+		<div class="topnav">
+			<a class="active" href="home.html">Home</a>
```
Comment (eskoglund) [inline home.html]: Ändra till "Hem" i texten eftersom vi kör allt annat på svenska

File: js/map/classes.js [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+    /**
+     * updateGuideLine - update the guideline coordinates to mouse coordinates
+     * 				   - from mousemove.
+     *
+     * @param {object} event
+     * @returns {void}
+     *
+     **/
+    updateGuideLine(event) {
+        // Behövs kommenteras
```
Comment (eskoglund) [inline js/map/classes.js]: Ta bort eller kommentera ifall det behövs

File: js/map/loadLeafletMaterialMap.js [hunk 6]
```diff
@@ -0,0 +1,204 @@
+/* global L */
+
+let polylines = L.layerGroup();
+let markers = L.layerGroup();
+let polygons = L.layerGroup();
+let newDiv;
+let boundsArray = [];
+// Imports Google maps javascript api key from getKey.js file
```
Comment (eskoglund) [inline js/map/loadLeafletMaterialMap.js]: Kan flyttas ner en rad

File: js/map/loadLeafletMaterialMap.js [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+    map.fitBounds(bounds);
+};
+
+map.on("zoomend", () => {
+    var i = 1;
+
+    markers.eachLayer((marker) => {
+        var pixelPosition = map.latLngToLayerPoint(marker._latlng);
+        //console.log(pixelPosition);
```
Comment (eskoglund) [inline js/map/loadLeafletMaterialMap.js]: Ta bort ifall den inte används

File: js/products/newObject.js [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+ *
+ * @returns {void}
+ */
+let newField = () => {
+    let div = document.createElement('div');
+
+    div.className = 'newField';
+    div.innerHTML =
+        `<input class="newKey" type="text" placeholder="titel">
+		<input class="newInput" type="text" placeholder="värde">
```
Comment (eskoglund) [inline js/products/newObject.js]: Tror det ska vara stor bokstav på "värde", samt möjligen "titel" på raden ovanför

File: js/products/updateObject.js [hunk 9]
```diff
@@ -0,0 +1,372 @@
...
+ *
+ * @returns {void}
+ */
+let newField = () => {
+    let div = document.createElement('div');
+
+    div.className = 'newField';
+    div.innerHTML =
+        `<input class="newKey" type="text" placeholder="titel">
+		<input class="newInput" type="text" placeholder="värde">
```
Comment (eskoglund) [inline js/products/updateObject.js]: Samma som kommentaren lite längre upp. Ge "värde" och "titel" stor bokstav

File: js/products/updateObject.js [hunk 10]
```diff
@@ -0,0 +1,372 @@
...
+		<input id="nrOfPumps" type="text" value="${nr}">`;
+
+    document.getElementById('Kategori').after(div);
+    let select = document.getElementById('pumpSelect');
+
+    for (let i = 0; i < pumps.length; i++) {
+        if (currentPump != pumps[i].Modell) {
+            let option = document.createElement('option');
+
+            //Modell i framtiden
```
Comment (eskoglund) [inline js/products/updateObject.js]: En svensk kommentar?

File: newObject.html [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+		<link href="https://fonts.googleapis.com/css?family=Fira+Sans:300,400" rel="stylesheet">
+		<link href="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.8.0/Chart.min.css" rel="stylesheet">
+		<link rel="stylesheet" href="css/newObject.css">
+		<script src="js/configVariables.js" charset="utf-8"></script>
+		<script src="js/products/newObject.js"></script>
+		<script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.8.0/Chart.min.js"></script>
+	</head>
+	<body>
+		<div class="topnav">
+			<a href="home.html">Home</a>
```
Comment (eskoglund) [inline newObject.html]: Ändra till "Hem" eftersom vi kör på svenska

File: updateObject.html [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+		<link href="https://fonts.googleapis.com/css?family=Fira+Sans:300,400" rel="stylesheet">
+		<link href="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.8.0/Chart.min.css" rel="stylesheet">
+		<link rel="stylesheet" href="css/updateObject.css">
+		<script src="js/configVariables.js" charset="utf-8"></script>
+		<script src="js/products/updateObject.js"></script>
+		<script src="https://cdnjs.cloudflare.com/ajax/libs/Chart.js/2.8.0/Chart.min.js"></script>
+	</head>
+	<body>
+		<div class="topnav">
+			<a href="home.html">Home</a>
```
Comment (eskoglund) [inline updateObject.html]: Ändra till "Hem" eftersom vi kör på svenska

File: updateObject.html [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+			<a href="calculations.html">Beräkningar</a>
+			<a class="active" href="listProducts.html">Produkter</a>
+			<a class="right" href="#logout">Logga ut</a>
+		</div>
+		<div class="wrap">
+			<div class="title">
+				<h1>Updatera Produkt</h1>
```
Comment (eskoglund) [inline updateObject.html]: Du har missat ett "p" i "Updatera Produkt". Det ska alltså stå "Uppdatera"

**Review by eskoglund — CHANGES_REQUESTED** [review]
I have made some comments on some small changes I want you to fix

**Review by eskoglund — APPROVED** [review]
Good job!

File: materiallist.html [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+<html lang="sv" dir="ltr">
+
+<head>
+	<meta charset="utf-8">
+	<title>Skriva ut</title>
+	<link rel="stylesheet" href="https://unpkg.com/leaflet@1.4.0/dist/leaflet.css" integrity="sha512-puBpdR0798OZvTTbP4A8Ix/l+A4dHDD0DGqYW6RQ+9jxkRFclaxxQb/SJAWZfWAkuyeQUytO7+7N4QKrDh+drA==" crossorigin=""/>
+	<link rel="stylesheet" href="css/materiallist.css">
+	<script src="https://unpkg.com/leaflet@1.4.0/dist/leaflet.js" integrity="sha512-QVftwZFqvtRNi0ZyCtsznlKSWOStnDORoefr1enyq5mVL4tmKB3S/EnC3rRJcxCPavG10IcrVGSmPh6Qw5lwrg==" crossorigin=""></script>
+	<script type="module" src="js/loadLeafletMaterialMap.js"></script>
+	<script type="module" src="js/getKey.js"></script>
```
Comment (SpaceLenore) [inline materiallist.html]: `js/getKey.js` is not found since it has been moved to `/map/`

File: materiallist.html [hunk 15]
```diff
@@ -0,0 +1,23 @@
+<!DOCTYPE html>
+<html lang="sv" dir="ltr">
+
+<head>
+	<meta charset="utf-8">
+	<title>Skriva ut</title>
+	<link rel="stylesheet" href="https://unpkg.com/leaflet@1.4.0/dist/leaflet.css" integrity="sha512-puBpdR0798OZvTTbP4A8Ix/l+A4dHDD0DGqYW6RQ+9jxkRFclaxxQb/SJAWZfWAkuyeQUytO7+7N4QKrDh+drA==" crossorigin=""/>
+	<link rel="stylesheet" href="css/materiallist.css">
+	<script src="https://unpkg.com/leaflet@1.4.0/dist/leaflet.js" integrity="sha512-QVftwZFqvtRNi0ZyCtsznlKSWOStnDORoefr1enyq5mVL4tmKB3S/EnC3rRJcxCPavG10IcrVGSmPh6Qw5lwrg==" crossorigin=""></script>
+	<script type="module" src="js/loadLeafletMaterialMap.js"></script>
```
Comment (SpaceLenore) [inline materiallist.html]: `js/loadLeafletMap.js` is not found since it has been moved to `/map/`

**Review by SpaceLenore — CHANGES_REQUESTED** [review]
As you can see by my comments, some js files are not found (404) due to the file structure has changed to use different folders. I found two but haven't had the possibility to look at all script tags. Please fix these two and verify the other script tags.

**Review by SpaceLenore — APPROVED** [review]

**Pull request merged**